rusoto/core/src/reactor.rs

```diff
@@ -102,9 +103,9 @@ impl Future for RequestDispatcherFuture {
 impl DispatchSignedRequest for RequestDispatcher {
     type Future = RequestDispatcherFuture;
 
-    fn dispatch(&self, signed_request: SignedRequest) -> Self::Future {
+    fn dispatch(&self, signed_request: SignedRequest, timeout: Option<Duration>) -> Self::Future {
         let (tx, rx) = oneshot::channel();
-        if let Some(err) = self.sender.unbounded_send((signed_request, tx)).err() {
+        if let Some(err) = self.sender.unbounded_send(((signed_request, timeout), tx)).err() {
             panic!("failed to send request to reactor: {}", err);
```

Comment:
This isn't in your change, but yikes. I didn't realize we panic'd here. One day it'd be nice to remove all `panic!` that could actually be hit.

Reply:
This'll panic only in the rare case where the reactor itself goes away (crashes), in which case a panic is appropriate imo.

As soon as we incorporate the new tokio crate we can remove all of this code anyway :)